### DISCUSSION TITLE
How to use the jovian kernel?

### DISCUSSION THREAD

**Discussion by pneg** [Jovian-Experiments/Jovian-NixOS#458] [Q&A] [answered]
I'm using Jovian as a flake. In my configuration.nix I have this:

```nix
boot.kernelPackages = inputs.jovian.legacyPackages."${pkgs.system}".linux_jovian;
```

This doesn't work because kernelPackages expects... something related to a set with a kernel package in it; I'm not sure what the documentation is trying to say.

How would I use the kernel with the SteamOS patches? Would I have to overlay the kernel package for the existing set of the specific kernel version in nixpkgs?

And how would I install additional modules? In the case if the version is no longer in cache would I have to compile any additional modules by myself?

**Comment by K900** [comment, marked as answer]
The modules set up the right kernel for you automatically.

**Reply by pneg** [reply]
It seems I forgot to rebuild it after changing my configuration. It is compiling now.
